pkg/local_workflows/filter_workflow.go

```diff
@@ -0,0 +1,152 @@
+	"github.com/snyk/go-application-framework/pkg/local_workflows/local_models"
+	"github.com/snyk/go-application-framework/pkg/utils"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+	"github.com/spf13/pflag"
+)
+
+const (
+	FilterFindingsWorkflowName = "findings.filter"
+)
+
+var WORKFLOWID_FILTER_FINDINGS = workflow.NewWorkflowIdentifier(FilterFindingsWorkflowName)
+
+func InitFilterFindingsWorkflow(engine workflow.Engine) error {
+	flags := pflag.NewFlagSet(FilterFindingsWorkflowName, pflag.ExitOnError)
+	_, err := engine.Register(WORKFLOWID_FILTER_FINDINGS, workflow.ConfigurationOptionsFromFlagset(flags), filterFindingsEntryPoint)
+
+	return err
+}
+
+type FindingsFilterFunc func(local_models.FindingResource) bool
+
+func filterSeverityASC(original []string, severityMinLevel string) []string {
+	if severityMinLevel == "" {
+		return original
+	}
+
+	minLevelPointer := slices.Index(original, severityMinLevel)
+
+	if minLevelPointer >= 0 {
+		return original[minLevelPointer:]
+	}
+
+	return original
+}
+
+func getSeverityThresholdFilter(severityThreshold string, severityOrder []string) FindingsFilterFunc {
+	return func(finding local_models.FindingResource) bool {
+		allowed_severities := filterSeverityASC(severityOrder, severityThreshold)
+
+		return utils.Contains(allowed_severities, string(finding.Attributes.Rating.Severity.Value))
+	}
+}
+
+func applyFilters(findingsModel *local_models.LocalFinding, filters []FindingsFilterFunc) {
+	// filteredFindings := local_models.LocalFinding{}
+	filteredFindings := []local_models.FindingResource{}
+	for _, finding := range findingsModel.Findings {
+		match := true
+		for _, filter := range filters {
+			if match {
+				match = filter(finding)
+			} else {
+				break
+			}
+		}
+		if match {
+			filteredFindings = append(filteredFindings, finding)
+		}
+	}
+	findingsModel.Findings = filteredFindings
+}
+
+func filterFindingsEntryPoint(invocationCtx workflow.InvocationContext, input []workflow.Data) (output []workflow.Data, err error) {
+	config := invocationCtx.GetConfiguration()
+	logger := invocationCtx.GetEnhancedLogger()
+	output = []workflow.Data{}
+
+	severityThreshold := config.GetString(configuration.FLAG_SEVERITY_THRESHOLD)
+
+	if severityThreshold == "" {
+		logger.Println("Severity threshold not set, skipping filter")
+		return input, nil
+	}
+
+	logger.Println("Workflow Name:", FilterFindingsWorkflowName)
+
+	for _, data := range input {
+		if strings.HasPrefix(data.GetContentType(), content_type.LOCAL_FINDING_MODEL) {
+			var findingsModel local_models.LocalFinding
+			findingsBytes, ok := data.GetPayload().([]byte)
+			if !ok {
+				var findingsError snyk_errors.Error = snyk_errors.Error{
+					Title:          "Invalid Payload Type",
+					Classification: "Internal",
+					Level:          "warning",
+					Detail:         "Invalid payload type",
+				}
+				logger.Err(findingsError).Msg(findingsError.Error())
```

Comment:
Suggestion: make the message more meaningful, the data object has information like the content location or the identifier, having this information in the logs can help trace the data through the system